OpenQA/Isotovideo/Utils.pm

```diff
@@ -53,6 +55,33 @@ sub git_remote_url ($git_repo_dir) {
     return 'UNKNOWN (error on git remote call)';    # uncoverable statement
 }
 
+sub _handle_caching ($clone_url, $clone_depth, $branch, $clone_cmd, $handle_output) {
+    # determine cache directory and ensure its parent directory exists
+    return undef unless my $git_cache_dir = $bmwqemu::vars{GIT_CACHE_DIR};
+    my $cache_dir = path($git_cache_dir, $clone_url->path);
+    path($git_cache_dir, $clone_url->path->to_dir)->make_path;
+
+    # lock the cache directory
+    open(my $lock, '>', "$cache_dir.lock") or die "Unable to open lock file '$cache_dir.lock' for Git caching: $!\n";
+    flock($lock, LOCK_EX) or die "Unable to lock '$cache_dir.lock' for Git caching: $!\n";
+    my $lock_guard = scope_guard sub {
+        flock($lock, LOCK_UN) or die "Unable to unlock '$cache_dir.lock' after Git caching: $!\n";
+        close($lock);
+    };
```

Comment:
blocks of codes preceeded by a comment and ending with a blank line IMHO are prime candidates for moving to functions:

```suggestion
sub _lock_cache_directory ($cache_dir) {
    open(my $lock, '>', "$cache_dir.lock") or die "Unable to open lock file '$cache_dir.lock' for Git caching: $!\n";
    flock($lock, LOCK_EX) or die "Unable to lock '$cache_dir.lock' for Git caching: $!\n";
    return scope_guard sub {
        flock($lock, LOCK_UN) or die "Unable to unlock '$cache_dir.lock' after Git caching: $!\n";
        close($lock);
    };
}
…
     my $lock_guard = _lock_cache_directory($cache_dir);
```